lib/spack/spack/test/cmd/mirror.py

```diff
@@ -48,6 +49,38 @@ def test_regression_8083(tmpdir, capfd, mock_packages, mock_fetch, config):
     assert 'as it is an external spec' in output
 
 
+@pytest.mark.regression('17531')
+def test_regression_17531(tmp_scope, tmpdir, capfd, mock_packages, mock_fetch, config):
```

Comment:
Can we turn this into a parametrized test (it seems it's effectively testing two cases)?

Reply:
I can break this up into two separate tests.  I think that makes more sense.